agent/command/assume_ec2_role.go

```diff
@@ -102,9 +100,7 @@ func (r *ec2AssumeRole) Execute(ctx context.Context,
 
 	creds := stscreds.NewCredentials(session1, r.RoleARN, func(arp *stscreds.AssumeRoleProvider) {
 		arp.RoleSessionName = strconv.Itoa(int(time.Now().Unix()))
-		if r.ExternalId != "" {
-			arp.ExternalID = utility.ToStringPtr(r.ExternalId)
-		}
+		arp.ExternalID = utility.ToStringPtr(fmt.Sprintf("%s-%s", conf.ProjectRef.Id, conf.Task.Requester))
```

Comment:
Is this format standardized by Build and will it always be in this format? If so, can we add a comment saying that we "just know" that it must have this format because it's been defined in an external trust policy?

Reply:
this is the format requested by build that was requested by cloud who *i think* would be the only users of this feature

Reply:
comment added though